apollo-federation/src/merge.rs

```diff
@@ -81,6 +92,33 @@ impl Debug for MergeFailure {
 }
 
 pub fn merge_subgraphs(subgraphs: Vec<&ValidSubgraph>) -> Result<MergeSuccess, MergeFailure> {
+    let mut merger = Merger::new();
+    let mut federation_subgraphs = ValidFederationSubgraphs::new();
+    for subgraph in subgraphs {
+        federation_subgraphs
+            .add(ValidFederationSubgraph {
+                name: subgraph.name.clone(),
+                url: subgraph.url.clone(),
+                schema: ValidFederationSchema::new(subgraph.schema.clone()).map_err(|e| {
+                    MergeFailure {
+                        schema: None,
+                        errors: vec![e.to_string()],
+                        composition_hints: Default::default(),
+                    }
+                })?,
+            })
+            .map_err(|e| MergeFailure {
+                schema: None,
+                errors: vec![e.to_string()],
+                composition_hints: Default::default(),
+            })?;
```

Comment:
There is a `From<FederationError>` implementation `MergeFailure`, so you can simply remove the boilerplate `map_err` implementations.

```suggestion
       federation_subgraphs.add(ValidFederationSubgraph {
            name: subgraph.name.clone(),
            url: subgraph.url.clone(),
            schema: ValidFederationSchema::new(subgraph.schema.clone())?,
        })?;
```

Reply:
good catch, thanks!